samples/interactivePersonalAccount.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * When using personal accounts with all of the authentication methods that accept a `domain` property in the optional parameters, if the `domain` is not specified, the credentials will end up not being able to access any of the resources of the personal account. For that same reason, the list of subscriptions expected in the return value of these methods will be empty.
+ *
+ * The workaround is to pass a known tenant Id to the `domain` property in the optional parameters when using these authentication methods. You can get the tenant Id from Azure portal or the Azure CLI.
+ *
+ * If you need to fetch the tenant Id programmatically:
+ *
+ * - Use any of the authentication methods without setting the domain to get a credential.
+ * - Call the `buildTenantLists(credential)` method by sending that same credential as the first parameter to get the list of all tenants in your account.
+ *
+ * If you want to avoid having to create a new credential, once you have access to one of the account's tenants, you can set it as the `domain` of your credentials by using the method `setDomain(tenant)` from the existing credentials object.
+ *
+ * This sample shows when the authentication fails for personal accounts, and how to overcome this issue.
+ *
+ * For context: https://github.com/Azure/ms-rest-nodeauth/issues/89#issuecomment-643471343
+ */
+import * as msRestNodeAuth from "../lib/msRestNodeAuth";
+import { SubscriptionClient } from "@azure/arm-subscriptions";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// Copy the content of "sample.env" to a file named ".env". It should be stored at the root of the repo.
+// then from the root of the cloned repo run, ts-node ./samples/interactive.ts
+
+async function main(): Promise<void> {
+  const authentication = await msRestNodeAuth.interactiveLoginWithAuthResponse();
+
+  // For personal accounts, the below will print an empty array as we did not set the domain.
+  console.log(
+    "Subscriptions retrieved by default",
+    authentication.subscriptions
+  );
+
+  const client = new SubscriptionClient(authentication.credentials);
+
+  // Note: You must replace `<my-subscription>` with the Id of one of your subscriptions to get the error.
+  // If you use this code as it is, you will get `The subscription '<my-subscription>' could not be found.`,
+  // which is not an error relevant to the issue we're describing.
```

Comment:
We shouldnt need to go into these details. The error message `The subscription '<my-subscription>' could not be found.` is self explanatory

```suggestion
  // Note: Replace `<my-subscription>` below with the Id of one of your subscriptions
```

Reply:
Why I think this is important:
The sample can be misleading if users run it without changing `<my-subscription>`. The sample won't show what is expected.